tests/storages_tests/test_storages.py

```diff
@@ -1238,7 +1238,7 @@ def failed_trial_callback(study: "optuna.Study", trial: FrozenTrial) -> None:
         trial = study.ask()
         trial.set_system_attr("test", "B")
         storage.record_heartbeat(trial._trial_id)
-        time.sleep(grace_period + 1)
+        time.sleep((heartbeat_interval * 2) + 1)
```

Comment:
This line does not refer to the value of `grace_period`. So, if we add the case `grace_period=4`, the test will fail.

Could you use the value of `grace_period` if it is not `None`? The following is an example:
```python
_grace_period = (heartbeat_interval * 2) if grace_period is None else grace_period
time.sleep(_grace_period + 1)
```

Reply:
Ok, fixed.